corehq/apps/app_manager/tests/test_extension_case.py

```diff
@@ -227,46 +242,32 @@ def test_xform_case_block_index_valid_relationship(self):
 
 class ExtensionCasesCreateOwnerID(SimpleTestCase):
 
-    def test_advanced_xform_create_owner_id_with_without_extensions(self):
-        """ Owner id should be automatically set if there are any non-extension indices"""
+    def test_advanced_xform_autoset_owner_id(self):
+        """
+            Owner id should be automatically set if there are any non-extension indices.
+            It should never be set if there are any dynamically-specified indices.
+        """
+
+        def _test_relationships(relationships, expected):
+            advanced_open_action = AdvancedOpenCaseAction.wrap({'case_indices': [{
+                'tag': 'tag{}'.format(int(random.random() * 10000)),
```

Comment:
Is it important for test passage that this tag value is unique? If so using a random value could result in a rare intermittently failing test if two of the random values picked turned out to be the same. Might be worth rearranging things a bit:
```suggestion
                'tag': 'tag{}'.format(i),
```

Reply:
Good point, will update in a followup PR.